### PR TITLE
Update replication_key state for descending streams

### PR DESCRIPTION
**Pull request by ericboucher**
A few limitations in the GitHub API force us to paginate a few streams in descending order. Since we cannot leverage the automated signpost updates from the SDK, this PR makes sure we properly update the state at the end of the run.

In addition, this PR switches the `commits` stream to use `use_fake_since_parameter` since this endpoint paginates in descending order.

### CODE REVIEW

**Review by laurentS — APPROVED** [review]

File: tap_github/client.py [hunk 1]
```diff
@@ -54,6 +54,32 @@ def http_headers(self) -> Dict[str, str]:
...
+        if self.use_fake_since_parameter:
+            state = self.get_context_state(context)
+            if set(["replication_key_signpost", "replication_key"]).issubset(
+                state.keys()
+            ):
+                record: Dict = {}
+                record[state["replication_key"]] = state["replication_key_signpost"]
+                self._increment_stream_state(
+                    latest_record=record,
+                    context=context,
```
Comment (edgarrmondragon) [inline tap_github/client.py:80]: @ericboucher this will not run until `get_records` is actually iterated over, given the _lazy_ nature of generators.

Comment (ericboucher) [reply tap_github/client.py:80]: You mean at the end of super.get_records()? If so that's the goal yes. Or am I missing something? And get_records() needs to be called in a special way? Should this live somewhere else then? @edgarrmondragon

**Comment by ericboucher** [comment]
I am not entirely sure it is working as expected and don't really have a good way to test. Open to ideas @laurentS @edgarrmondragon :)

**Comment by laurentS** [comment]
> I am not entirely sure it is working as expected and don't really have a good way to test. Open to ideas @laurentS @edgarrmondragon :)

@ericboucher I've added a test for your code. I *think* it does what you want. It's a bit clunky as I had to fake both the system time and the response from github, but might be a useful example for other corner cases we need to verify in the future. Feel free to adjust if I've missed anything!

**Comment by laurentS** [comment]
This PR is probably not needed anymore with meltano/sdk#1164

**Comment by ericboucher** [comment]
Solved by #164
